tests/unit/fragment_array_property_test.js

```diff
@@ -140,20 +140,25 @@ module('unit - `MF.fragmentArray` property', function(hooks) {
         region: 'Crownlands',
         country: 'Westeros'
       });
-      assert.deepEqual(person.changedAttributes().addresses, undefined, 'a frehsly pushed object has no changes');
+      assert.deepEqual(person.changedAttributes().addresses, undefined, 'a freshly pushed object has no changes');
+      assert.ok(!person._internalModel.hasChangedAttributes(), 'a freshly pushed object has no changed attributes');
```

Comment:
Does `hasDirtyAttributes` give us the same results as `person._internalModel.hasChangedAttributes()`?

Reply:
I don't even know. I think it should by reading the [documentation](https://guides.emberjs.com/v3.22.0/models/creating-updating-and-deleting-records/)
But `hasChangedAttributes` is an internal method, and this is the one we need to update for fixing the bug in my case.
As `hasChangedAttributes` will be called as I developed it in earlier post.